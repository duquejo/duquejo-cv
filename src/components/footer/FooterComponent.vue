<template>
  <footer>
    <ul v-if="props.socialLinks" class="flex gap-x-4">
      <li v-for="link in props.socialLinks" :key="link.name" class="icon">
        <a :href="link.url" target="_blank" :title="link.title">
          <v-icon :name="link.icon" scale="1"/>
        </a>
      </li>
    </ul>
    <span>©️ {{ currentYear }} José Duque</span>
  </footer>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import { SocialLinks } from '@/interfaces/social-links';

interface Props {
  socialLinks?: SocialLinks[];
}
const props = defineProps<Props>();

const currentYear = computed(() => new Date().getFullYear() );
</script>