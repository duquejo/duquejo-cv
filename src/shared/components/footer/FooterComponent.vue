<template>
  <footer>
    <ul v-if="SOCIAL_LINKS" class="flex gap-x-4">
      <li v-for="link in SOCIAL_LINKS" :key="link.name" class="icon">
        <a :href="link.url" target="_blank" :title="link.title">
          <v-icon :name="link.icon" scale="1"/>
        </a>
      </li>
      <!-- <li class="icon">
        <router-link :to="{ name: 'contact' }">
          <v-icon name="md-email-outlined" scale="1"/>
        </router-link>
      </li> -->
    </ul>
    <span>©️ {{ currentYear }} José Duque</span>
  </footer>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import { social as SOCIAL_LINKS } from '../../locale/es-CO/config.json';
const currentYear = computed(() => new Date().getFullYear() );
</script>

<style scoped>
</style>