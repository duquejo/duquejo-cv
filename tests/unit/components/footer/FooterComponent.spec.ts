import { shallowMount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import FooterComponent from '../../../../src/components/footer/FooterComponent.vue';
import { SocialLinksBuilder } from '../../../utils/SocialLinksBuilder';

describe('+ FooterComponent.vue tests', () => {

  const socialLinks = new SocialLinksBuilder().withDefaultLinks().build();

  const wrapper = shallowMount(FooterComponent, {
    props: {
      socialLinks,
    },
  });


  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render the social links if they are given', () => {
    const li = wrapper.findAll('li');
    expect(li.length).toBe(socialLinks.length);
  });

  it('Should validate if social links are provided', () => {
    const wrapperWithoutLinks = shallowMount(FooterComponent);
    const li = wrapperWithoutLinks.findAll('li');
    expect(li.length).toBe(0);
  });

  it('Should generate the current year automatically', () => {
    const span = wrapper.find('span');
    expect(span.text()).toContain(new Date().getFullYear());
  });
});