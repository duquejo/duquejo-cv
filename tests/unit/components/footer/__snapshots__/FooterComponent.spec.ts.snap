// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ FooterComponent.vue tests > Should match the snapshot 1`] = `
"<footer>
  <ul class="flex gap-x-4">
    <li class="icon"><a href="https://github.com" target="_blank" title="gitHub">
        <v-icon name="bi-github" scale="1"></v-icon>
      </a></li>
    <li class="icon"><a href="https://linkedin.com" target="_blank" title="linkedIn">
        <v-icon name="bi-linkedin" scale="1"></v-icon>
      </a></li>
  </ul><span>©️ 2024 José Duque</span>
</footer>"
`;
